.eslintrc.json

```diff
@@ -71,6 +71,8 @@
       ],
       "rules": {
         "comma-dangle": 0,
+        "no-unused-expressions": 0,
+        "@typescript-eslint/no-unused-expressions": 2,
```

Comment:
Why are we allowing for unused expressions? This seems like there's an issue with the code, not the eslint config.

Reply:
This isn't allowing for unused expressions, it's disabling for Typescript files the default linting rule for unused expressions and enabling the Typescript version.

The default JS rule accuses the function calls with optional chaining of being an unused expression: `error  Expected an assignment or function call and instead saw an expression  no-unused-expressions`

Reply:
Here is the relevant bit of typescript-eslint documentation: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-expressions.md

Reply:
@devinbinnie I could revert back to the previous options if this is really an issue, but I'd also have to remove the optional chaining function calls that you suggested on a previous review.

Reply:
@brunoro Optional chaining should work as expected without the need for this rule. We did have an issue that was caused by an older version of `typescript-eslint` that was recently rectified here: https://github.com/mattermost/mattermost-webapp/pull/5195

Reply:
@devinbinnie I just tested it again after merging master back and it didn't help much with the issue. Could you elaborate on the downsides of disabling the JS `no-unused-expressions` rule in favor of the typescript specific one?

Reply:
Maybe I'm just failing to identify the issues on the code. Would you lend another pair of eyes? Here are the lines accused by the linter as unused expressions, without the TS-specific rule:
* https://github.com/mattermost/mattermost-webapp/blob/9202add3ee63cd9e93ef7092af2a563e15eb8687/components/textbox/textbox.tsx#L121
* https://github.com/mattermost/mattermost-webapp/blob/9202add3ee63cd9e93ef7092af2a563e15eb8687/components/textbox/textbox.tsx#L147
* https://github.com/mattermost/mattermost-webapp/blob/9202add3ee63cd9e93ef7092af2a563e15eb8687/components/textbox/textbox.tsx#L151
* https://github.com/mattermost/mattermost-webapp/blob/9202add3ee63cd9e93ef7092af2a563e15eb8687/components/textbox/textbox.tsx#L155
* https://github.com/mattermost/mattermost-webapp/blob/9202add3ee63cd9e93ef7092af2a563e15eb8687/components/textbox/textbox.tsx#L159
* https://github.com/mattermost/mattermost-webapp/blob/9202add3ee63cd9e93ef7092af2a563e15eb8687/components/textbox/textbox.tsx#L163
* https://github.com/mattermost/mattermost-webapp/blob/9202add3ee63cd9e93ef7092af2a563e15eb8687/components/textbox/textbox.tsx#L182
* https://github.com/mattermost/mattermost-webapp/blob/9202add3ee63cd9e93ef7092af2a563e15eb8687/components/textbox/textbox.tsx#L186
* https://github.com/mattermost/mattermost-webapp/blob/9202add3ee63cd9e93ef7092af2a563e15eb8687/components/textbox/textbox_links/textbox_links.tsx#L27

Reply:
You're using it to conditionally call a method that has no expression to put its return value into, hence why TypeScript thinks you're defining an expression that is unused. While this is a valid use case of the operator, I don't think the correct move here is to change the lint configuration, which is there to protect our code from having unused variables from being defined.

I would instead for the cases above, just do a basic null check on the parent before calling the method rather than use the optional chaining operator.

Reply:
@devinbinnie I think you're not understanding what the proposed ESLint rule changes entail. The default JS `no-unused-expressions` rule is being replaced by `@typescript-eslint/no-unused-expressions` rule still protects the code, while linting corretcly the (afaik) valid usage of the operator in Typescript. Is there any downside of using the `@typescript-eslint` version I'm not aware of?

Reply:
> Here is the relevant bit of typescript-eslint documentation: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-expressions.md

@devinbinnie please refer to the documentation link I attached some days ago in this thread.

Reply:
Sorry, I misread what you were trying to accomplish. You can go ahead and replace the rule with the `typescript-eslint` version. Apologies for the confusion.